sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheckTest.java

```diff
@@ -0,0 +1,154 @@
+        checkConfig.addAttribute("methodName", "exit2");
+        checkConfig.addAttribute("argumentCount", "");
+
+        final String[] expected = {
+            "31:31: " + getCheckMessage(MSG_KEY_WITHOUT_ARG, "exit2", "exit2"),
+            "32:40: " + getCheckMessage(MSG_KEY_WITHOUT_ARG, "exit2", "exit2"),
+        };
+        verify(checkConfig, getPath("InputForbidCertainMethodsCheck.java"), expected);
+    }
+
+    @Test
+    public void testWithRegexForMethodName() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ForbidCertainMethodsCheck.class);
+        checkConfig.addAttribute("methodName", "assert(True|False)");
+        checkConfig.addAttribute("argumentCount", "1");
+
+        final String[] expected = {
+            "29:26: " + getCheckMessage(MSG_KEY_WITH_ARG, "assertTrue", "assert(True|False)", 1,
+                "1"),
+        };
+        verify(checkConfig, getPath("InputForbidCertainMethodsCheck.java"), expected);
+    }
+
+    @Test
+    public void testUnsupportedToken() {
+        final DetailAST sync = new DetailAST();
+        sync.setType(TokenTypes.LITERAL_SYNCHRONIZED);
+        try {
+            final ForbidCertainMethodsCheck check = new ForbidCertainMethodsCheck();
+            check.visitToken(sync);
+            fail("Expecting IllegalArgumentException");
+        }
+        catch (IllegalArgumentException ex) {
+            Assert.assertEquals("Found unsupported token: LITERAL_SYNCHRONIZED", ex.getMessage());
+        }
+    }
+
+    @Test
+    public void testNullMethodNameRegex() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ForbidCertainMethodsCheck.class);
+        checkConfig.addAttribute("methodName", null);
+        verify(checkConfig, getPath("InputForbidCertainMethodsCheck.java"), new String[] {});
+    }
+
+    @Test
+    public void testEmptyMethodNameRegex() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ForbidCertainMethodsCheck.class);
+        checkConfig.addAttribute("methodName", "");
+        try {
+            verify(checkConfig, getPath("InputForbidCertainMethodsCheck.java"), new String[] {});
+        }
+        catch (CheckstyleException ex) {
+            checkExceptionMessage(ex,
+                "cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - "
+                    + "Cannot set property 'methodName' to '' in module "
+                    + "com.github.sevntu.checkstyle.checks.coding.ForbidCertainMethodsCheck");
+        }
+    }
+
+    @Test
+    public void testBadMethodNameRegex() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ForbidCertainMethodsCheck.class);
+        checkConfig.addAttribute("methodName", "[exit");
+        try {
+            verify(checkConfig, getPath("InputForbidCertainMethodsCheck.java"), new String[] {});
```

Comment:
This is a new check that isn't in yet. All try/catch in tests should have a fail as the last statement in the try.

Reply:
Done.